src/utils/daemon.rs

```diff
@@ -11,39 +11,37 @@ use crate::{
 use chrono::{Timelike, Utc};
 use failure::Error;
 use log::{debug, error, info};
-use std::path::PathBuf;
-use std::sync::Arc;
-use std::time::Duration;
-use std::{env, thread};
-
-fn start_registry_watcher(pool: Pool, build_queue: Arc<BuildQueue>) -> Result<(), Error> {
-    thread::Builder::new()
-        .name("registry index reader".to_string())
-        .spawn(move || {
-            // space this out to prevent it from clashing against the queue-builder thread on launch
-            thread::sleep(Duration::from_secs(30));
-            loop {
-                let opts = opts();
-                let mut doc_builder = DocBuilder::new(opts, pool.clone(), build_queue.clone());
-
-                if doc_builder.is_locked() {
-                    debug!("Lock file exists, skipping checking new crates");
-                } else {
-                    debug!("Checking new crates");
-                    match doc_builder.get_new_crates() {
-                        Ok(n) => debug!("{} crates added to queue", n),
-                        Err(e) => error!("Failed to get new crates: {}", e),
-                    }
-                }
+use std::{env, path::PathBuf, sync::Arc};
+use tokio::{
+    task::{self, JoinHandle},
+    time::{self, Duration, Instant},
+};
 
-                thread::sleep(Duration::from_secs(60));
-            }
-        })?;
+async fn start_registry_watcher(pool: Pool, build_queue: Arc<BuildQueue>) -> JoinHandle<()> {
+    task::spawn(async move {
+        let mut interval = time::interval(Duration::from_secs(60));
 
-    Ok(())
+        let opts = opts();
+        let mut doc_builder = DocBuilder::new(opts, pool.clone(), build_queue.clone());
+
+        loop {
+            interval.tick().await;
```

Comment:
60 seconds between runs. Otherwise you might start simultaneous builds by accident, and we don't have enough hardware for that (or fine-grained locking, etc).

Reply:
Well, it's still serial, so it would just not pause at all if the last run took over 60 seconds (and I don't remember how `Interval` works if you miss an interval, I _think_ it doesn't back up, and essentially just pauses till you poll it again).

Reply:
Oh I see. Starting the next build immediately sounds fine.